src/Cake.Common/Solution/Project/Properties/AssemblyInfoCreatorData.cs

```diff
@@ -94,9 +99,17 @@ private void AddAttribute(string name, string @namespace, string value)
             }
         }
 
-        private void AddCustomAttribute(string name, string @namespace, string value)
+        private void AddCustomAttribute(string name, string @namespace, object value)
         {
-            if (value != null)
+            if (value is bool)
+            {
+                AddAttributeCore(CustomAttributes, name, @namespace, (bool)value ? _trueStringValue : _falseStringValue);
+            }
+            else if (value == null || string.IsNullOrEmpty(value as string))
```

Comment:
This should be `else if (value == null || (value is string stringValue && string.IsNullOrEmpty(stringValue))` otherwise we will never reach the `else` clause.

Reply:
IMHO a pattern switch would be cleaner something like
```csharp
        private void AddCustomAttribute(string name, string @namespace, object value)
        {
            var attributeValue = AttributeValueToString(value);

            AddAttributeCore(CustomAttributes, name, @namespace, attributeValue);
        }

        private string AttributeValueToString(object value)
        {
            switch (value)
            {
                case null:
                {
                    return string.Empty;
                }
                case bool boolValue:
                {
                    return boolValue ? _trueStringValue : _falseStringValue;
                }
                case string stringValue:
                {
                    return stringValue == string.Empty
                        ? string.Empty
                        : string.Concat("\"", value, "\"");
                }
                default:
                {
                    return Convert.ToString(value, CultureInfo.InvariantCulture);
                }
            }
        }
```

also should non strings be quoted? I.e. int/long/etc? In example above I to string object invariant.  To be more robust should we escape `"` like `\"`?

Reply:
Also unrelated `name` in `AddMetadataAttribute(string name, string @namespace, string key, string value)` doesn't seem to be use, it's private so wouldn't break anything to remove it it it's indeed not used.

Reply:
I agree, a switch statement here makes a lot of sense.  I will change it up to use this.  Honestly, I am not sure regarding handling for int/longs, etc.  As I mentioned in the commits/comments, this change wasn't an attempt to handle all conditions, but rather the ones that I know currently are prevented.  I would suggest that we handle any edge cases on a case by case basis.  Thoughts?

Reply:
We can create another issue for the name property, as this is not related to this PR.

Reply:
>  As I mentioned in the commits/comments, this change wasn't an attempt to handle all conditions, but rather the ones that I know currently are prevented. I would suggest that we handle any edge cases on a case by case basis. Thoughts?

Then I think it makes more sense in default switch case to either not quote and use invariant as my example above, or throw a not supported exception.